### PR TITLE
New adapter madvertise

### PR DESCRIPTION
**Pull request by NLopezMad**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

New bidder adapter (madvertise)

- test parameters for validating bids
size: [320, 50]
```
{
  bidder: 'madvertise',
  params: {
    s: "/4543756/prebidadaptor/madvertiseHB"
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- nicolas.lopez@madvertise.com
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by NLopezMad** [comment]
Hello, 
Can i have an update on the progress made with the review please ? The github bot seems to view this pull request as abandonned.

Thanks

**Comment by NLopezMad** [comment]
Hello,

We see "needs docs" label. it seems https://github.com/prebid/prebid.github.io/pull/623 is good. What is missing on doc ?

**Comment by NLopezMad** [comment]
Ok, therefore our adaptor will be available on next realease of prebid (1.6) ?

**Comment by NLopezMad** [comment]
Hello,

There isn't LGTM label on our adaptor ?

File: modules/madvertiseBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+// use protocol relative urls for http or https
+const MADVERTISE_ENDPOINT = 'https://mobile.mng-ads.com/';
+
+export const spec = {
+  code: 'madvertise',
+  aliases: ['madvertise'],
```
Comment (mkendall07) [inline modules/madvertiseBidAdapter.js]: please remove this. You can't alias your own adapter with the same name.

File: modules/madvertiseBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  /**
+   * @param {object} bid
+   * @return boolean
+   */
+  isBidRequestValid: function (bid) {
+    if (typeof bid.params !== 'object') {
+      return false;
+    }
+    if (!Array.isArray(bid.sizes) || bid.sizes.length == 0) {
```
Comment (mkendall07) [inline modules/madvertiseBidAdapter.js]: we have a utility function to parse the bid.sizes. See https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L114

File: modules/madvertiseBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  for (var i = 0; i < cookies.length; i++) {
+    var cookie = cookies[i].trim();
+    if (cookie.match(/seenad\[\d+\]=\d+/)) {
+      src += '&' + cookie;
+    }
+  }
+  return src;
+}
+
+function _setSeenAd(adid) {
```
Comment (mkendall07) [inline modules/madvertiseBidAdapter.js]: why are you saving this data in the publisher domain?

**Comment by mkendall07** [comment]
@NLopezMad sorry for the delay. Please update the PR so we can merge. Thanks

**Comment by NLopezMad** [comment]
Hello,

I have updated PR according your feedback.

File: modules/madvertiseBidAdapter.js (hunk 3 above)
Comment (NLopezMad) [reply modules/madvertiseBidAdapter.js]: For capping, we send adId display on device. We need to remove this feature ?

**Comment by NLopezMad** [comment]
Sorry I have pushed a fix for testing and parseSizesInput

File: modules/madvertiseBidAdapter.js (hunk 3 above)
Comment (NLopezMad) [reply modules/madvertiseBidAdapter.js]: it's ok or not for _setSeenAd ?

Comment (mkendall07) [reply modules/madvertiseBidAdapter.js]: Please remove it for now

Comment (NLopezMad) [reply modules/madvertiseBidAdapter.js]: done.

File: modules/madvertiseBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+   */
+  isBidRequestValid: function (bid) {
+    if (typeof bid.params !== 'object') {
+      return false;
+    }
+    let sizes = utils.parseSizesInput(bid.sizes);
+    if (!sizes || sizes.length === 0) {
+      return false;
+    }
+    console.log(sizes[0]);
```
Comment (mkendall07) [inline modules/madvertiseBidAdapter.js]: remove please.

Comment (NLopezMad) [reply modules/madvertiseBidAdapter.js]: I have removed console.log

**Comment by NLopezMad** [comment]
I have removed console.log

**Comment by NLopezMad** [comment]
thx !

**Pull request merged**